### PR TITLE
courses: smoother texts (fixes #3814)

### LINKED ISSUE #3814: courses text warnings

### PR DESCRIPTION
**Pull request by ewhang5**
fixes #3814

### CODE REVIEW

File: app/src/main/java/org/ole/planet/myplanet/base/BaseContainerFragment.kt [hunk 1]
```diff
@@ -69,7 +69,7 @@ abstract class BaseContainerFragment : BaseResourceFragment() {
 
     fun setRatings(`object`: JsonObject?) {
         if (`object` != null) {
-            AdapterCourses.showRating(`object`, rating, timesRated, ratingBar)
+            AdapterCourses.showRating(`object`, rating, timesRated, ratingBar, context)
```
Comment (Okuro3499) [inline app/src/main/java/org/ole/planet/myplanet/base/BaseContainerFragment.kt:72]: I don't think it is necessary to pass context here

**Review by Okuro3499 — CHANGES_REQUESTED** [review]
I get this crash
[Screen_recording_20240711_141123.webm](https://github.com/open-learning-exchange/myplanet/assets/64019708/af3ce398-cb0e-4756-bc20-132abcf9356a)

```
FATAL EXCEPTION: main
Process: org.ole.planet.myplanet, PID: 24126
java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{org.ole.planet.myplanet/org.ole.planet.myplanet.ui.courses.CourseProgressActivity}: java.lang.InstantiationException: java.lang.Class<org.ole.planet.myplanet.ui.courses.CourseProgressActivity> has no zero argument constructor
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3689)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3922)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:139)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:96)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2443)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at android.app.ActivityThread.main(ActivityThread.java:8177)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
Caused by: java.lang.InstantiationException: java.lang.Class<org.ole.planet.myplanet.ui.courses.CourseProgressActivity> has no zero argument constructor
	at java.lang.Class.newInstance(Native Method)
	at android.app.AppComponentFactory.instantiateActivity(AppComponentFactory.java:95)
	at androidx.core.app.CoreComponentFactory.instantiateActivity(CoreComponentFactory.java:44)
	at android.app.Instrumentation.newActivity(Instrumentation.java:1378)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3676)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3922) 
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103) 
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:139) 
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:96) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2443) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:205) 
	at android.os.Looper.loop(Looper.java:294) 
	at android.app.ActivityThread.main(ActivityThread.java:8177) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971) 

```

File: app/src/main/java/org/ole/planet/myplanet/base/BaseContainerFragment.kt (hunk 1 above)
Comment (ewhang5) [reply app/src/main/java/org/ole/planet/myplanet/base/BaseContainerFragment.kt:72]: Okay I will look into this

Comment (ewhang5) [reply app/src/main/java/org/ole/planet/myplanet/base/BaseContainerFragment.kt:72]: @Okuro3499 I ended up passing in context to use the getString() method. I think the crash is related to something else, so I will be fixing that. Can you let me know if you still think I should not pass context there?

**Review by Okuro3499 — APPROVED** [review]

**Review by dogi — APPROVED** [review]

**Review by dogi — APPROVED** [review]

**Pull request merged**